pkg/driver/driver.go

```diff
@@ -35,120 +31,69 @@ type Driver interface {
 	// The second argument may contain error occurred while starting driver
 	Start(_ context.Context) (chan error, error)
 
-	// CanRunGUI returns bool to indicate if the hostagent need to run GUI synchronously
-	CanRunGUI() bool
-
-	// RunGUI is for starting GUI synchronously by hostagent. This method should be wait and return only after vm terminates
-	// It returns error if there are any failures
-	RunGUI() error
-
 	// Stop will terminate the running vm instance.
 	// It returns error if there are any errors during Stop
 	Stop(_ context.Context) error
-
-	// Register will add an instance to a registry.
-	// It returns error if there are any errors during Register
-	Register(_ context.Context) error
-
-	// Unregister will perform any cleanup related to the vm instance.
-	// It returns error if there are any errors during Unregister
-	Unregister(_ context.Context) error
-
-	ChangeDisplayPassword(_ context.Context, password string) error
-
-	GetDisplayConnection(_ context.Context) (string, error)
-
-	CreateSnapshot(_ context.Context, tag string) error
-
-	ApplySnapshot(_ context.Context, tag string) error
-
-	DeleteSnapshot(_ context.Context, tag string) error
-
-	ListSnapshots(_ context.Context) (string, error)
-
-	// ForwardGuestAgent returns if the guest agent sock needs forwarding by host agent.
-	ForwardGuestAgent() bool
-
-	// GuestAgentConn returns the guest agent connection, or nil (if forwarded by ssh).
-	GuestAgentConn(_ context.Context) (net.Conn, error)
-}
-
-type BaseDriver struct {
-	Instance *store.Instance
-
-	SSHLocalPort int
-	VSockPort    int
-	VirtioPort   string
 }
 
-var _ Driver = (*BaseDriver)(nil)
-
-func (d *BaseDriver) Validate() error {
-	return nil
-}
-
-func (d *BaseDriver) Initialize(_ context.Context) error {
-	return nil
-}
-
-func (d *BaseDriver) CreateDisk(_ context.Context) error {
-	return nil
-}
-
-func (d *BaseDriver) Start(_ context.Context) (chan error, error) {
-	return nil, nil
-}
-
-func (d *BaseDriver) CanRunGUI() bool {
-	return false
-}
-
-func (d *BaseDriver) RunGUI() error {
-	return nil
-}
+// GUI defines GUI-related operations.
+type GUI interface {
+	// RunGUI is for starting GUI synchronously by hostagent. This method should be wait and return only after vm terminates
+	// It returns error if there are any failures
+	RunGUI() error
 
-func (d *BaseDriver) Stop(_ context.Context) error {
-	return nil
+	ChangeDisplayPassword(ctx context.Context, password string) error
+	GetDisplayConnection(ctx context.Context) (string, error)
```

Comment:
In Go "getters" don't use a `Get` prefix. You know it is a getter because the method name doesn't include an action word (verb).

```suggestion
	DisplayConnection(ctx context.Context) (string, error)
```